src/Stoa.ts

```diff
@@ -448,11 +445,9 @@ class Stoa extends WebService
                     else if (UInt64.compare(height.value, expected_height.value) > 0)
                     {
                         // Recovery is required for blocks that are not received.
-                        let success: boolean = await this.recoverBlock(block, height, expected_height);
-                        if (!success)
-                        {
-                            this.pool.unshift(stored_data);
-                        }
+                        let success: boolean = false;
+                        while (!success)
+                            success = await this.recoverBlock(block, height, expected_height);
```

Comment:
I'm not 100% sure of this retry logic though. It matches the previous retry logic, but we should re-evaluate this in the future.